submarine-server/server-core/src/main/java/org/apache/submarine/server/JobManager.java

```diff
@@ -45,10 +47,9 @@
 
   private final AtomicInteger jobCounter = new AtomicInteger(0);
 
-  private final ConcurrentMap<JobId, Job> jobs = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, JobSpec> cachedJobSpecs = new ConcurrentHashMap<>();
```

Comment:
Could you please put a comment here explaining the meaning of the key in cachedJobSpecs? Adding a comment to the key struct will make it easier to read.

Reply:
Done